base_optional_quick_create/models/ir_model.py

```diff
@@ -50,4 +50,7 @@ def create(self, vals_list):
     def write(self, vals):
         res = super().write(vals)
         self._patch_quick_create()
+        if "avoid_quick_create" in vals:
+            self.pool.registry_invalidated = True
+            self.pool.signal_changes()
```

Comment:
Isn't it enough to set `registry_invalidated`? I think any `write` call will be wrapped in `manage_changes` here: https://github.com/odoo/odoo/blob/13.0/odoo/service/model.py#L40, so that `signal_changes` is called here: https://github.com/odoo/odoo/blob/13.0/odoo/modules/registry.py#L507, and that method will indeed undertake action when `registry_invalidated` is set (here: https://github.com/odoo/odoo/blob/13.0/odoo/modules/registry.py#L475).

Reply:
Oops links are to 13.0 but same code can be found in Odoo 12.0, just on different line numbers.

Reply:
Hi, thanks for having a look at this!
Unfortunately I'm looking at all your links after some time so the line numbers are quite different.
Anyway I looked into it and just invalidating the registry is enough for triggering a signal to other workers because the response computation is all wrapped in a web request context manager (https://github.com/odoo/odoo/blob/716f36b172c4e129f2ba049ae33faa7d42fc500e/odoo/http.py#L1467-L1492) and it signals for changes on exit (https://github.com/odoo/odoo/blob/716f36b172c4e129f2ba049ae33faa7d42fc500e/odoo/http.py#L287).

This signaling is very far from the code where we need the signaling to be done (this PR's code) and I'd rather keep the signaling here too because if the signaling is removed from https://github.com/odoo/odoo/blob/716f36b172c4e129f2ba049ae33faa7d42fc500e/odoo/http.py#L287 then it would be very hard to find out why this module stopped working.
Also, I think having the signaling in here is more explicit on the fact the we need the signaling to be done after this `write`, given the condition `"avoid_quick_create" in vals`.

TL;DR: you are right, but I'd rather keep it like this.

Reply:
OK, thanks for giving it a thought.